### ISSUE TITLE
Operations on integers and strings with a BigInt should return a BigInt

### ISSUE THREAD

**Issue by faheel** [faheel/BigInt#2]
Performing binary operations on integers or strings with a `BigInt` object as the second operand should return a `BigInt`:
```C++
big_int_1 =  5  + big_int_2;
big_int_1 = "5" + big_int_2;
// similarly for all other binary arithmetic and relational operators
```

**Comment by faheel** [comment]
I'm working on it

**Comment by faheel** [comment]
Closed in commits e7bcce9 and c5a9e77.
